torch/lib/c10d/ProcessGroupMPI.cpp

```diff
@@ -91,6 +92,74 @@ void checkSameSizeAndType(
   }
 }
 
+void checkSplitSizes(
+    const std::vector<int64_t>& split_sizes,
+    const at::Tensor& tensor,
+    int group_size) {
+  if (split_sizes.size() == 0) {
+    if (tensor.size(0) % group_size != 0) {
```

Comment:
Let's use `TORCH_CHECK(pred, message)` for user-facing error checking, and `TORCH_INERNAL_ASSERT(pred, message)` for developer-facing error checking. There two are more preferable in recent PyTorch releases.